src/routes/chat/Chat.jsx

```diff
@@ -34,10 +35,22 @@ const ChatContent = () => {
 };
 
 const Chat = () => {
+  const dispatch = useDispatch();
   const { chatRoomId } = useParams();
   const correspondentName = useSelector(
     state => state.chat.chatRoom.length > 0 && state.chat.chatRoom[0].dealerName
   );
+  const { openChatSocket, closeChatSocket } = useSocket();
+  useEffect(() => {
+    openChatSocket();
```

Comment:
Repeating: we don't initialize this here, we should initialize them in `AuthenticatedApp` component, and close the socket there too